src/commands/upgrade/upgrade.py

```diff
@@ -0,0 +1,116 @@
+from logging import getLogger
+from pathlib import Path
+import shutil
+import os
+from urllib.request import urlretrieve
+import tarfile
+
+from packaging import version
+import requests
+import tomli
+
+logger = getLogger()
+
+PROTOSTAR_REPO = "https://github.com/software-mansion/protostar"
+PROTOSTAR_DIR = Path.home() / ".protostar"
+
+
+def upgrade() -> None:
+    manager = UpgradeManager(PROTOSTAR_DIR)
+    manager.upgrade()
+
+
+class UpgradeManagerException(Exception):
+    pass
+
+
+class UpgradeManager:
+    def __init__(self, protostar_dir: Path):
+        assert os.path.isdir(protostar_dir)
+        assert os.path.isdir(protostar_dir / "dist")
+        self.protostar_dir = protostar_dir
+        self.old_version = protostar_dir / "previous_version_tmp"
+
+        platform = self.get_platform()
+        self.tarball_name = f"protostar-{platform}.tar.gz"
+        self.tarball_loc = PROTOSTAR_DIR / self.tarball_name
+
+        self.current_version = self.get_current_version()
+        self.latest_version_tag = self.get_latest_release()["tag_name"]
+        self.latest_version = version.parse(self.latest_version_tag)
+
+    def is_current_latest(self):
+        return self.latest_version == self.current_version
+
+    def upgrade(self):
+        logger.info("Looking for a new version ...")
+        if self.latest_version <= self.current_version:
```

Comment:
I am not sure whether this condition will always work as expected e.g. "0.0.10" < "0.0.9"

Reply:
We do not compare them as strings but as parsed version objects

Reply:
Yep, my bad. I checked the type of `latest_version_tag` rather than `latest_version`.